libraries/terminal/src/MockWritable.ts

```diff
@@ -1,17 +1,29 @@
 // Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
 // See LICENSE in the project root for license information.
 
-import { ITerminalChunk } from './ITerminalChunk';
+import { ITerminalChunk, TerminalChunkKind } from './ITerminalChunk';
 import { TerminalWritable } from './TerminalWritable';
-import { AnsiEscape } from '@rushstack/node-core-library';
+import { AnsiEscape, ITerminalProvider, TerminalProviderSeverity } from '@rushstack/node-core-library';
 
 /**
  * A {@link TerminalWritable} subclass for use by unit tests.
  *
  * @beta
  */
-export class MockWritable extends TerminalWritable {
+export class MockWritable extends TerminalWritable implements ITerminalProvider {
```

Comment:
This is mixing together two currently unrelated APIs.

The `StringBufferTerminalProvider` from `node-core-library` already does most of this for the `ITermalProvider` interface.

Reply:
I need a way to route the output from a `Terminal` and a `TerminalWritable` to the same backing store for evaluating the correctness of the output. If you have another avenue to suggest for doing so, I'm open to ideas.